scripts/sanitize.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Canonical Ltd.
+# Generate a Javascript file that can be used to sanitize a Mongo database so
+# it can be shared.
+from __future__ import print_function
+
+import sys
+
+# This lists the collections and fields in those collections that need to be sanitized
+to_sanitize = [
+ ('users', ['passwordhash']),
+ ('units', ['passwordhash']),
+ ('machines', ['passwordhash']),
+ ('settings', ['settings']),
+ ('controllers', ['settings', 'cert', 'privatekey', 'caprivatekey', 'sharedsecret', 'systemidentity']),
+ ('actions', ['message', 'results']),
+ ('cloudCredentials', ['attributes']),
+]
+
+def generateScript():
+    lines = []
+    # Create an index on the transactions so that the updates go faster
+    lines.append('print(new Date().toLocaleString())')
+    lines.append('db.txns.createIndex({"o.c": 1})')
+    for collection, attributes in to_sanitize:
+        # First we generate the sanitization of the collection itself
+        # (we don't use .format() because {} is used all the time in Javascript
+        lines.append('print(new Date().toLocaleString())')
+        lines.append('print("updating collection %s for %s")' % (collection, attributes))
+        lines.append('print(db.%s.update({}, {"$set": {' % (collection,))
+        for attribute in attributes:
+            lines.append('    "{}": "REDACTED",'.format(attribute))
+        # replace the last ',' with a '}'
+        lines[-1] = lines[-1][:-1] + '}'
```

Comment:
yeah, I didn't mean to use .format(). I started trying to use it, but it turned out I had to put { and {} in lots of the rest of the code.

I do wish Javascript supported the trailing comma:
[
  foo,
  bar,
]

(Maybe javascript does but JSON doesn't? regardless changing syntax for the last entry sucks)
(done)